### ISSUE TITLE
[Module Proposal]: `avm-ptn-openai-cognitivesearch-tf`

### ISSUE THREAD

**Issue by mikestiers** [Azure/Azure-Verified-Modules#1312]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Terraform

### Module Classification?

Pattern Module

### Module Name

avm-ptn-openai-cognitivesearch

### Module Details

This is a pattern module proposal for a Corporate Line of Business (LoB) ChatBot application using Azure OpenAI and Azure Cognitive Search Service. It includes:

- Azure App Gateway
- Azure App Service Premium v3 (Linux): Web App + API App
- Azure BLOB Storage
- Azure SQL Database
- Azure Cache for REDIS
- Azure App Insights
- Private VNET

*Note* This is taking over from https://github.com/Azure/Azure-Verified-Modules/issues/1000

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

mikestiers

### (Optional) Secondary Module Owner's GitHub Username (handle)

andbron

**Comment by prjelesi** [comment]
Hi @mikestiers 

since this is request for pattern module, we would need few more things to check:

- Where is it coming from/what’s backing it - e.g., Azure Architecture Center (AAC), community request, customer example. 
- Provide an architectural diagram and related documentation if possible - or a pointer to these if they are publicly available.

Tnx!

**Comment by mikestiers** [comment]
I'm working on the architecture diagram. I thought we had it already, but I just checked and it's not there so I'll reply on Thursday hopefully with it.

In the meantime, this architecture is coming from a generic customer example that has been provided by Architect @mariocon. We have quite a few patterns that we will be contributing that are a result of customer examples gathered by @mariocon from a number of customer engagements.

**Comment by prjelesi** [comment]
@mikestiers 

Thanks for requesting/proposing to be an AVM module owner!

We just want to confirm **you agree to the below pages** that define what module ownership means:

- [Team Definitions & RACI](https://azure.github.io/Azure-Verified-Modules/specs/shared/team-definitions)
- [Shared Specification (Bicep & Terraform)](https://azure.github.io/Azure-Verified-Modules/specs/shared)
- [Module Support](https://azure.github.io/Azure-Verified-Modules/help-support/module-support)

Any questions or clarifications needed, let us know!

If you agree, please just **reply to this issue with the exact sentence below** (as this helps with our automation 👍):

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

Thanks,

The AVM Core Team

#RR

**Comment by mikestiers** [comment]
I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER

**Comment by prjelesi** [comment]
@mikestiers 


Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

Before starting development, please ensure ALL the following requirements are met.

**Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:

- For your module:
  - `ModuleName` - for naming your module
  - `TelemetryIdPrefix` - for your module's [telemetry](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr3---category-telemetry---deploymentusage-telemetry)
- For your module's repository:
  - Repo name and folder path are defined in `RepoURL`
  - Create GitHub teams for module owners and contributors and grant them permissions as outlined [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only).
  - Grant permissions for the AVM core team and PG teams on your GitHub repo as described [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions).

Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍

You can now start the development of this module! ✅ Happy coding! 🎉

**Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team

**Comment by mikestiers** [comment]
Hello here is a pull request that requires review: https://github.com/Azure/terraform-azurerm-avm-ptn-openai-cognitivesearch/pull/15

**Comment by jeanchg** [comment]
Development on hold for the moment, until #2113 is complete. It will then resume and be updated accordingly.

**Comment by mbilalamjad** [comment]
@jeanchg https://github.com/Azure/Azure-Verified-Modules/issues/2113 is currently in development and requestor of this module is part of https://github.com/Azure/Azure-Verified-Modules/issues/2113

@mikestiers do you believe this is required given the work we're doing on AI Landing Zone?
